controllers/backstage_route.go

```diff
@@ -56,9 +56,57 @@ func (r *BackstageReconciler) applyBackstageRoute(ctx context.Context, backstage
 		}
 	}
 
+	r.applyRouteParamsFromCR(route, backstage)
+
 	err = r.Create(ctx, route)
 	if err != nil {
 		return fmt.Errorf("failed to create backstage route, reason: %s", err)
 	}
 	return nil
 }
+
+func (r *BackstageReconciler) applyRouteParamsFromCR(route *openshift.Route, backstage bs.Backstage) {
+	if backstage.Spec.Application == nil || backstage.Spec.Application.Route == nil {
+		return // Nothing to override
+	}
+	routeCfg := backstage.Spec.Application.Route
+	if len(routeCfg.Host) > 0 {
```

Comment:
nit: this seems more legible to me, as it makes it clear that `routeCfg.Host` is a `string`, not a slice.

```suggestion
	if routeCfg.Host != "" {
```

Reply:
Ok - fine then.